core/matching/json_path_match.go

```diff
@@ -1,19 +1,47 @@
 package matching
 
-import "github.com/NodePrime/jsonpath"
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/client-go/util/jsonpath"
+)
 
 func JsonPathMatch(matchingString string, toMatch string) bool {
-	paths, err := jsonpath.ParsePaths(matchingString)
+	matchingString = prepareJsonPathQuery(matchingString)
+
+	jsonPath := jsonpath.New("")
+
+	err := jsonPath.Parse(matchingString)
 	if err != nil {
 		return false
```

Comment:
It'll be good to add some debug logs before each return statements in this method. It will help to understand why the match failed, whether is invalid jsonpath expression or cannot unmarshal the `toMatch` string